pylint/testutils/_primer/primer_compare_command.py

```diff
@@ -0,0 +1,142 @@
+# Licensed under the GPL: https://www.gnu.org/licenses/old-licenses/gpl-2.0.html
+# For details: https://github.com/PyCQA/pylint/blob/main/LICENSE
+# Copyright (c) https://github.com/PyCQA/pylint/blob/main/CONTRIBUTORS.txt
+from __future__ import annotations
+
+import json
+from pathlib import Path
+
+from pylint.testutils._primer.primer_command import PackageMessages, PrimerCommand
+
+MAX_GITHUB_COMMENT_LENGTH = 65536
+
+
+class CompareCommand(PrimerCommand):
+    def run(self) -> None:
+        main_dict = self._load_json(self.config.base_file)
+        new_dict = self._load_json(self.config.new_file)
+        final_main_dict = self._cross_reference(main_dict, new_dict)
```

Comment:
I don't really like how `_cross_reference` modifies `new_dict` in place and we then pass the same object to `_create_comment`. Could we return two objects from `_cross_reference` and then pass those?
In general we seem to avoid modifying arguments to functions and then re-using them later on.

Reply:
b1aa1c22a